shoplist/src/main/java/com/netguru/codereview/ui/model/ResultState.kt

```diff
@@ -0,0 +1,6 @@
+package com.netguru.codereview.ui.model
+
+sealed  class ResultState<T> {
+    data class Success<T>(val data: T): ResultState<T>()
+    data class Failed<T>(val error: Throwable? = null) : ResultState<T>()
+}
```

Comment:
notify UI about the error if it happened, https://developer.android.com/kotlin/coroutines/coroutines-best-practices#exceptions